tools/eth1exporter/main.go

```diff
@@ -0,0 +1,173 @@
+// Prometheus exporter for Ethereum address balances.
+// Forked from https://github.com/hunterlong/ethexporter
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"math/big"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var (
+	allWatching []*Watching
+	loadSeconds float64
+	totalLoaded int64
+	eth         *ethclient.Client
+)
+
+var (
+	port            = flag.Int("port", 9090, "Port to serve /metrics")
+	web3URL         = flag.String("web3-url", "https://goerli.prylabs.net", "Web3 URL to access information about ETH1")
+	prefix          = flag.String("prefix", "", "Metrics prefix.")
+	addressFilePath = flag.String("addresses", "", "File path to addresses text file.")
+)
+
+func main() {
+	flag.Parse()
+
+	if *addressFilePath == "" {
+		log.Println("--addresses is required")
+		return
+	}
+
+	err := OpenAddresses(*addressFilePath)
+	if err != nil {
+		panic(err)
+	}
+
+	err = ConnectionToGeth(*web3URL)
+	if err != nil {
+		panic(err)
+	}
+
+	// check address balances
+	go func() {
+		for {
+			totalLoaded = 0
+			t1 := time.Now()
+			fmt.Printf("Checking %v wallets...\n", len(allWatching))
+			for _, v := range allWatching {
+				v.Balance = GetEthBalance(v.Address).String()
+				totalLoaded++
+			}
+			t2 := time.Now()
+			loadSeconds = t2.Sub(t1).Seconds()
+			fmt.Printf("Finished checking %v wallets in %0.0f seconds, sleeping for %v seconds.\n", len(allWatching), loadSeconds, 15)
+			time.Sleep(15 * time.Second)
+		}
+	}()
+
+	block := CurrentBlock()
+
+	fmt.Printf("ETHexporter has started on port %v using web3 server: %v at block #%v\n", *port, *web3URL, block)
+
+	http.HandleFunc("/metrics", MetricsHTTP)
+	http.HandleFunc("/reload", ReloadHTTP)
+	panic(http.ListenAndServe(fmt.Sprintf("0.0.0.0:%d", *port), nil))
```

Comment:
why a panic here ? won't this crash the whole routine

Reply:
yes, we should panic if we can't serve the webserver